client-app/src/examples/todo/TaskDialogModel.js

```diff
@@ -0,0 +1,103 @@
+import {HoistModel, managed} from '@xh/hoist/core';
+import {
+    FormModel,
+    lengthIs,
+    required
+} from '@xh/hoist/cmp/form';
+import {observable, action, makeObservable} from '@xh/hoist/mobx';
+
+export class TaskDialogModel extends HoistModel {
+
+    /** @member {TodoPanelModel} */
+    parentModel;
+
+    @observable
+    isAdd = false;
+
+    @observable
+    isOpen = false;
+
+    @managed
+    formModel = new FormModel({
+        fields: [
+            {
+                name: 'id'
+            },
+            {
+                name: 'complete'
+            },
+            {
+                name: 'description',
+                rules: [required, lengthIs({max: 1000})]
+            },
+            {
+                name: 'dueDate',
+                displayName: 'Due Date'
+            }
+        ]
+    });
+
+    constructor(todoPanelModel) {
+        super();
+        makeObservable(this);
+
+        this.parentModel = todoPanelModel;
+    }
+
+    reset() {
+        this.formModel.reset();
+    }
+
+    async submitAsync() {
+        const {formModel, parentModel} = this,
+            {values} = formModel,
+            isValid = await formModel.validateAsync();
+
+        if (isValid) {
+            const {description, dueDate, complete} = values,
+                existingId = values.id,
+                task = {
+                    id: existingId ?? Date.now(),
```

Comment:
what is our standard here for generating unique ids on client -- this was my quick idea.